packages/clerk-js/src/core/auth/AuthCookieService.ts

```diff
@@ -66,6 +72,11 @@ export class AuthCookieService {
     private instanceType: InstanceType,
     private clerkEventBus: ReturnType<typeof createClerkEventBus>,
   ) {
+    // Create shared lock for cross-tab token refresh coordination.
+    // This lock is used by both the poller and the focus handler to prevent
+    // concurrent token fetches across tabs.
+    this.tokenRefreshLock = SafeLock(REFRESH_SESSION_TOKEN_LOCK_KEY);
```

Comment:
🤔  I'm thinking we should add the lock in `getToken()` itself, otherwise we'll have to patch any other places `getToken()` is used, and users won't have this taken care of out of the box